protocol/src/reqres/reqresapi.rs

```diff
@@ -0,0 +1,56 @@
+use crate::reqres::*;
+use jsonrpc_core::Result;
+use jsonrpc_derive::rpc;
+use std::fmt;
+
+fn get_addr(address: String) -> (String,String) {
+  let ma = address.clone();
+  let v: Vec<&str> = address.split('/').collect();
+  let peer_id = v.last().unwrap().to_string();
+  
+  let mut multi_addr = ma.replace(&peer_id.clone(),"");
+  let _ = multi_addr.pop();
+  (multi_addr,peer_id)
+}
+
+#[rpc]
+pub trait  ReqRespT {
+    #[rpc(name = "p2p_get")]
+    /// Returns for p2p rpc get responce    
+    fn p2p_get(&self, address: String, message: String) -> Result<String> ;
+    #[rpc(name = "p2p_ping")]
+    /// Returns for reqresp p2p rpc ping responce
+    fn p2p_ping(&self, address: String) -> Result<String> ;
+}
+
+pub struct ReqRespApi;
+
+impl ReqRespT for ReqRespApi {
+
+    fn p2p_get(&self, address: String, message: String) -> Result<String> {
+        let (multiaddr, peerid) = get_addr (address.clone());
+        let value = Some(message.clone().to_string()); 
+        let method = "get".to_string();   
+
+        let res = reqresapi::reqres(multiaddr.clone(), peerid.clone(), method.clone(), value.clone()).unwrap();
+
+        let mut line = String::new();
+        println!("sent: {} {} {}",multiaddr.clone(), peerid, method);
+        fmt::write (&mut line, format_args!("{} {} {} {} {:?}",multiaddr.clone(), peerid, method, message, res)).unwrap_or(());
+      
+        Ok(line)
+  }
+
```

Comment:
Let’s report response delay as duration type in case of ping.

Reply:
added response delay in us  (dt as delta time) to formatted sting:

        println!("sent: {} {} {} {} us",multiaddr.clone(), peerid, ping.clone(), dt);
        fmt::write (&mut line, format_args!("{} {} {} {:?} {} us",multiaddr.clone(), peerid, ping, res, dt)).unwrap_or(());
      
        Ok(line)